tfutils/base.py

```diff
@@ -469,31 +495,43 @@ def predict(step, results):
     return preds
 
 
-def get_valid_results(sess, valid_targets, save_intermediate_freq=None, dbinterface=None):
+def run_targets(sess, dbinterface, target_name, target, num_steps, 
+                online_agg_func, agg_func, save_intermediate_freq, validation_only):
+    """TODO:  this code resembles train() function, possible want to unify
+    """
+    agg_res = None
+    if save_intermediate_freq:
+        n0 = len(dbinterface.outrecs)
+    for _step in range(num_steps):
+        res = sess.run(target)
+        assert hasattr(res, 'keys'), 'result must be a dictionary'
+        if save_intermediate_freq and (_step % save_intermediate_freq == 0):
+            dbinterface.save({}, {target_name: res}, step=_step, validation_only=validation_only)
+        agg_res = online_agg_func(agg_res, res, _step)
+    result = agg_func(agg_res)
+    if save_intermediate_freq:
+        dbinterface.sync_with_host()
+        n1 = len(dbinterface.outrecs)    
+        result['intermediate_steps'] = dbinterface.outrecs[n0: n1]
+    return result
+
+
+def run_targets_dict(sess, targets, save_intermediate_freq=None, dbinterface=None, validation_only=False):
     """
     Helper function for actually computing validation results.
     """
-    valid_results = {}
-    for targname in valid_targets:
-        num_steps = valid_targets[targname]['num_steps']
-        targ = valid_targets[targname]['targets']
-        agg_func = valid_targets[targname]['agg_func']
-        online_agg_func = valid_targets[targname]['online_agg_func']
-        agg_res = None
-        if save_intermediate_freq:
-            n0 = len(dbinterface.outrecs)
-        for _step in range(num_steps):
-            res = sess.run(targ)
-            assert hasattr(res, 'keys'), 'result of validation must be a dictionary'
-            if save_intermediate_freq and (_step % save_intermediate_freq == 0):
-                dbinterface.save({}, {targname: res}, step=_step)
-            agg_res = online_agg_func(agg_res, res, _step)
-        valid_results[targname] = agg_func(agg_res)
-        dbinterface.sync_with_host()
-        if save_intermediate_freq:
-            n1 = len(dbinterface.outrecs)    
-            valid_results[targname]['intermediate_steps'] = dbinterface.outrecs[n0: n1]
-    return valid_results
+    results = {}
+    for target_name in targets:
+        num_steps = targets[target_name]['num_steps']
+        target = targets[target_name]['targets']
+        agg_func = targets[target_name]['agg_func']
+        online_agg_func = targets[target_name]['online_agg_func']
+        results[target_name] = run_targets(sess, dbinterface, target_name, target,
```

Comment:
I want to keep run_targets separated out as a function until we figure out how/whether to integrate it with train().